src/main/java/de/dennisguse/opentracks/content/TrackDataHub.java

```diff
@@ -412,8 +412,10 @@ private void notifyTrackPointsTableUpdate(boolean updateSamplingState, Set<Track
             lastSeenTrackPointId = localLastSeenTrackPointIdId;
         }
 
-        for (TrackDataListener listener : sampledInListeners) {
-            listener.onNewTrackPointsDone(trackPoint);
+        if (trackPoint != null) {
```

Comment:
This works but speed maintains the last speed if OT loses the GPS signal instead of resetting the speed to 0.

Look at this (screenshots when OT loses GPS - the speed should be 0):

![imagen](https://user-images.githubusercontent.com/4819612/110143130-2c52da00-7dd7-11eb-9a12-c090ec09d99f.png)

Reply:
Valid point. Have not thought about this one...

Reply:
After 5min thinking about it: I am not sure if this is an issue.
I mean, if I set the GPS recording time interval to 30min, what speed do I expect after 5min?
Is it the speed of the initial GPS reading or 0?

On the other hand, we have TrackPoint.isRecent() with a fixed timeout of 1 minute.
Anyhow, this functionality must be triggered by the _updateUIeachSecond_-Runnable.

Reply:
For me is not a problem, I mean this doesn't disturb me but I thought it could be an actual issue.
So, okay for me :clap:

Reply:
And what you say has a lot of sense ;)

Reply:
I added a commit that should (not tested it), reset the speed if the trackpoint is outdated (set it to 0) and if sensor data is present set it to it's value.

Reply:
I'm running it and I get this error when I try to start a new record:

```
2021-03-05 18:41:48.849 22986-23014/de.dennisguse.opentracks.debug E/eglCodecCommon: GoldfishAddressSpaceHostMemoryAllocator: ioctl_ping failed for device_type=5, ret=-1
2021-03-05 18:42:04.963 22986-22986/de.dennisguse.opentracks.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: de.dennisguse.opentracks.debug, PID: 22986
    java.lang.RuntimeException: Unable to create service de.dennisguse.opentracks.services.TrackRecordingService: java.lang.NullPointerException: Attempt to invoke virtual method 'long de.dennisguse.opentracks.content.data.Track$Id.getId()' on a null object reference
        at android.app.ActivityThread.handleCreateService(ActivityThread.java:3544)
        at android.app.ActivityThread.access$1300(ActivityThread.java:199)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1666)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'long de.dennisguse.opentracks.content.data.Track$Id.getId()' on a null object reference
        at de.dennisguse.opentracks.content.provider.ContentProviderUtils.getTrack(ContentProviderUtils.java:205)
        at de.dennisguse.opentracks.services.TrackRecordingService.restartTrackAfterServiceRestart(TrackRecordingService.java:347)
        at de.dennisguse.opentracks.services.TrackRecordingService.onCreate(TrackRecordingService.java:153)
```

Reply:
Forget what I wrote (I'v just deleted it)... Android Studio got messy or something like this.

All okay.

Reply:
Just out of curiosity: why TrackPoint.MAX_LOCATION_AGE == 1 minute? Is there science behind this?

Reply:
No clue, but I guess it had.
But as the knowledge was lost - we can do whatever we believe is the right choice ;)